### PR TITLE
Add File Uploader

### PR DESCRIPTION
**Pull request by artyymcflyy**
Add the ability to any files, at scale, for any machine learning model

- Upload files to any object storage solution
- Download object storage metadata
- Modify any custom dataset(from a NoSQLDatabase) Use that metadata by

### CODE REVIEW

File: src/file-uploader/update_object_permissions.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    s3_client = session.client('s3', endpoint_url=CONFIG["ENDPOINT_URL"])
+    print("Setup Comeplete")
+
+    print("Creating Object Client")
+    object_endpoint = CONFIG["OBJECT_CDN_URL"]
+    object_client = ObjectClient(s3_client,CONFIG["BUCKET_NAME"], object_endpoint)
+    print("Object Client created")
+
+    print("Updating Object permissions...")
+    object_client.update_object_permissions(CONFIG["OBJECT_ACL"])
```
Comment (jaded-gloryy) [inline src/file-uploader/update_object_permissions.py:17]: What are permissions needed for?

File: src/file-uploader/add_file_url.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    print("Setting up Database Connection")
+    # TODO: Add check for successful connection
+    mongo_client = MongoClient(CONFIG["CONNECTION_STRING"])
+    print("Connection setup")
+    print("Creating Database Client")
+    db_client = NoSQLDatabaseClient(mongo_client, CONFIG["DATABASE_NAME"])
+    print("Database Client created")
+
+    db_client.update_documents("plant_labels", [{"file_name": "121205.jpg", "url": "http://test.io/121205.jpg"}])
```
Comment (jaded-gloryy) [inline src/file-uploader/add_file_url.py:18]: Is this test code that should be removed?

File: src/file-uploader/upload_files.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import os
```
Comment (jaded-gloryy) [inline src/file-uploader/upload_files.py:1]: I'm not sure how this function works. It seems like upload_files reads files in a directory and get's their url, but I don't see an "action" after that. Is the command_string what does the uploading?

**Review by jaded-gloryy — COMMENTED** [review]
Please see the comments I left.

File: src/file-uploader/update_object_permissions.py (hunk 1 above)
Comment (artyymcflyy) [reply src/file-uploader/update_object_permissions.py:17]: All of the objects were uploaded as "private". In order for the images to be seen/downloaded all of the objects permissions needed to be updated to "public". This function runs 1 time and makes the change to every object in the storage

File: src/file-uploader/upload_files.py (hunk 3 above)
Comment (artyymcflyy) [reply src/file-uploader/upload_files.py:1]: That's a great question. The function relies on a bash scrip to use a command from the aws s3 CLI. An advantage for using the CLI command is that it handles looking for files within an unlimited number of directories. If I used the boto3 python library then I would have to write the logic to recursively check each folder and upload the objects individually.

Comment (artyymcflyy) [reply src/file-uploader/upload_files.py:1]: I also went and updated the functions and classes with docstrings to explain their purpose

**Review by jaded-gloryy — APPROVED** [review]

**Pull request merged**